main/src/main/scala/sbt/Defaults.scala

```diff
@@ -407,6 +409,12 @@ object Defaults extends BuildCommon {
       )
     },
     fileConverter := MappedFileConverter(rootPaths.value, allowMachinePath.value),
+    sourcePositionMappers ++= {
+      val fc = fileConverter.value
+      if (reportAbsolutePath.value) {
+        List(toAbsoluteSourceMapper(fc))
+      } else Nil
+    },
```

Comment:
@adpi2 @eed3si9n Setting a default for `sourcePositionMappers` is a very bad idea and breaks existing apps, see #6352